### PR TITLE
[openssl] Update openssl 1.1.1g - continued from PR #11003

### LINKED ISSUE #11127: [openssl] update to 1.1.1g - CVE-2019-1563 & CVE-2019-1551
**Library name**:
openssl

**Ports affected**:
openssl, openssl-unix, openssl-uwp, openssl-windows and any dependents (_Qt5-* for example_.)

**New version number**:
1.1.1g

**Other information that may be useful** (_release notes, etc..._)
Current version in vcpkg is 1.1.1d.
Since that release there have been CVE-2019-1563 & CVE-2019-1551 found.
https://www.openssl.org/news/vulnerabilities-1.1.1.html
The current advice is to update to 1.1.1g where ever possible.
openssl 1.1.1g was released 2020-Apr-21.
https://www.openssl.org/source/

Thanks!

### PR DESCRIPTION
**Pull request by Septarius**
**Describe the pull request**
Applies the suggestions from the review of PR #11003 to a rebased branch based off of PR #11003.

- Included the original three commits by @crackedmind and then did the rebase off upstream master.

- Applied the suggestions included in the original PR. 

- Additionally removed vcpkg_test_cmake(PACKAGE_NAME OpenSSL MODULE) from ports/openssl-uwp/portfile.cmake since it was review as deprecated in the other two portfile.cmake files.

- What does your PR fix? 
Fixes #11003 
Fixes #11127

- Which triplets are supported/not supported? Have you updated the CI baseline? I don't believe there should be any changes here.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? It should now.

### CODE REVIEW

**Comment by Septarius** [comment]
### Console log for x64_linux regression
There were no log files in the artifact even though it mentions them in the console log as being at:   /mnt/vcpkg-ci/buildtrees/apr-util/build-x64-linux-dbg-err.log

It failed on `ports/apr-util/portfile.cmake:78` `vcpkg_install_make()`
```
2020-06-11T05:05:08.9279850Z Starting package 6/449: apr-util:x64-linux
2020-06-11T05:05:08.9280356Z Building package apr-util[core]:x64-linux...
2020-06-11T05:05:08.9280799Z Could not locate cached archive: /archives/5d/5d121f9e4dd32ff8ddf098d9185938b2ba8ebdd3.zip
2020-06-11T05:05:13.4321382Z -- Downloading http://archive.apache.org/dist/apr/apr-util-1.6.1.tar.bz2...
2020-06-11T05:05:13.4322328Z -- Extracting source /mnt/vcpkg-ci/downloads/apr-util-1.6.1.tar.bz2
2020-06-11T05:05:13.4323007Z -- Using source at /mnt/vcpkg-ci/buildtrees/apr-util/src/apr-util-1-a62665826e
2020-06-11T05:05:13.4323527Z -- Configuring apr-util
2020-06-11T05:05:13.4323978Z -- Configuring x64-linux-dbg
2020-06-11T05:05:13.4324420Z -- Configuring x64-linux-rel
2020-06-11T05:05:13.4324865Z -- Building x64-linux-dbg
2020-06-11T05:05:13.4325246Z CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
2020-06-11T05:05:13.4325838Z     Command failed: /usr/bin/make V=1 -j 17 -f Makefile all
2020-06-11T05:05:13.4326460Z     Working Directory: /mnt/vcpkg-ci/buildtrees/apr-util/x64-linux-dbg
2020-06-11T05:05:13.4326846Z     See logs for more information:
2020-06-11T05:05:13.4327389Z       /mnt/vcpkg-ci/buildtrees/apr-util/build-x64-linux-dbg-err.log
2020-06-11T05:05:13.4327683Z 
2020-06-11T05:05:13.4327941Z Call Stack (most recent call first):
2020-06-11T05:05:13.4328317Z   scripts/cmake/vcpkg_build_make.cmake:200 (vcpkg_execute_build_process)
2020-06-11T05:05:13.4328713Z   scripts/cmake/vcpkg_install_make.cmake:24 (vcpkg_build_make)
2020-06-11T05:05:13.4329285Z   ports/apr-util/portfile.cmake:78 (vcpkg_install_make)
2020-06-11T05:05:13.4329645Z   scripts/ports.cmake:76 (include)
2020-06-11T05:05:13.4329882Z 
2020-06-11T05:05:13.4330055Z 
2020-06-11T05:05:13.5229116Z Error: Building package apr-util:x64-linux failed with: BUILD_FAILED
2020-06-11T05:05:13.5230591Z Elapsed time for package apr-util:x64-linux: 4.593 s
```

### Log of local Ubuntu 20.04 build
```
Elapsed time for package openssl:x64-linux: 11.95 ms
Starting package 5/5: apr-util:x64-linux
Building package apr-util[core]:x64-linux...
-- Downloading http://archive.apache.org/dist/apr/apr-util-1.6.1.tar.bz2...
-- Extracting source /home/vcpkgbuilder/vcpkg/downloads/apr-util-1.6.1.tar.bz2
-- Using source at /home/vcpkgbuilder/vcpkg/buildtrees/apr-util/src/apr-util-1-a62665826e
-- Configuring apr-util
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Installing x64-linux-dbg
-- Installing x64-linux-rel
-- Installing: /home/vcpkgbuilder/vcpkg/packages/apr-util_x64-linux/share/apr-util/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package apr-util[core]:x64-linux... done
Installing package apr-util[core]:x64-linux...
Installing package apr-util[core]:x64-linux... done
Elapsed time for package apr-util:x64-linux: 8.844 s
```
**Full log here**
[https://gist.github.com/Septarius/694106b0a9c43e696a5e93e6fc0e543c](https://gist.github.com/Septarius/694106b0a9c43e696a5e93e6fc0e543c)

**Comment by Septarius** [comment]
### Console log for x64_osx regression
There were no log files in the artifact even though it mentions them in the console log as being at:    /Users/vagrant/Data/buildtrees/apr-util/build-x64-osx-dbg-err.log

It failed on `ports/apr-util/portfile.cmake:78` `vcpkg_install_make()`

```
2020-06-11T05:09:17.5313600Z Starting package 6/400: apr-util:x64-osx
2020-06-11T05:09:17.5314230Z Building package apr-util[core]:x64-osx...
2020-06-11T05:09:17.5314700Z Could not locate cached archive: /Users/vagrant/Data/archives/c0/c011c9247600d1c57afefaa9cdd97fcd57a53ab3.zip
2020-06-11T05:09:48.0279690Z -- Using cached /Users/vagrant/Data/downloads/apr-util-1.6.1.tar.bz2
2020-06-11T05:09:48.0280740Z -- Extracting source /Users/vagrant/Data/downloads/apr-util-1.6.1.tar.bz2
2020-06-11T05:09:48.0281260Z -- Using source at /Users/vagrant/Data/buildtrees/apr-util/src/apr-util-1-a62665826e
2020-06-11T05:09:48.0281660Z -- Configuring apr-util
2020-06-11T05:09:48.0282110Z -- apr-util requires pkg-config from the system package manager (example: "sudo apt-get install pkg-config")
2020-06-11T05:09:48.0282520Z -- Configuring x64-osx-dbg
2020-06-11T05:09:48.0282930Z -- Configuring x64-osx-rel
2020-06-11T05:09:48.0283320Z -- Building x64-osx-dbg
2020-06-11T05:09:48.0283500Z CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
2020-06-11T05:09:48.0284670Z     Command failed: /usr/bin/make V=1 -j 6 -f Makefile all
2020-06-11T05:09:48.0285410Z     Working Directory: /Users/vagrant/Data/buildtrees/apr-util/x64-osx-dbg
2020-06-11T05:09:48.0285720Z     See logs for more information:
2020-06-11T05:09:48.0286710Z       /Users/vagrant/Data/buildtrees/apr-util/build-x64-osx-dbg-err.log
2020-06-11T05:09:48.0286960Z 
2020-06-11T05:09:48.0287100Z Call Stack (most recent call first):
2020-06-11T05:09:48.0287220Z   scripts/cmake/vcpkg_build_make.cmake:200 (vcpkg_execute_build_process)
2020-06-11T05:09:48.0287360Z   scripts/cmake/vcpkg_install_make.cmake:24 (vcpkg_build_make)
2020-06-11T05:09:48.0287760Z   ports/apr-util/portfile.cmake:78 (vcpkg_install_make)
2020-06-11T05:09:48.0289050Z   scripts/ports.cmake:76 (include)
2020-06-11T05:09:48.0289210Z 
2020-06-11T05:09:48.0289350Z 
2020-06-11T05:09:48.5137930Z Error: Building package apr-util:x64-osx failed with: BUILD_FAILED
2020-06-11T05:09:48.5139270Z Elapsed time for package apr-util:x64-osx: 31 s
```

**Comment by Septarius** [comment]
So I am not sure the cause of this build failure. The console output isn't very helpful tracking it down and we are missing the CI's make command's log files.

The only obvious difference in the CI log vs the local log is the local GCC is 9.3 vs 7.5 .

**Comment by Septarius** [comment]
Rebased to fix conflict with PR #11344. Will see if the CI build succeeds now.

**Comment by Septarius** [comment]
The x64_linux regressions are gone we have new ones in there place.

### arm64_windows regressions
`soundtouch` and `tinyfiledialogs`
Should be fixed by PR #11896

### x86_windows regressions
`geographiclib`
Should be fixed by PR #11896

### x64_osx regressions
Improvement over the last regression we have error logs in the [artifact](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/38704/artifacts?artifactName=x64-osx%20port%20build%20failure%20logs&api-version=5.1&%24format=zip)

**config-x64-osx-dbg-err.log**
```
configure: WARNING: unrecognized options: --disable-silent-rules, --enable-static, --disable-shared
./../src/apr-util-1-a62665826e/configure: line 4089: cd: /Volumes/data/work/2/s/installed/x64-osx/build-1: No such file or directory
cp: /apr_rules.mk: No such file or directory
configure: WARNING: unrecognized options: --disable-silent-rules, --enable-static, --disable-shared
```

**build-x64-osx-dbg-err.log**
```
Makefile:50: /Volumes/data/buildtrees/apr-util/x64-osx-dbg/build/rules.mk: No such file or directory
make: *** No rule to make target `/Volumes/data/buildtrees/apr-util/x64-osx-dbg/build/rules.mk'.  Stop.
```

I don't believe that the changes in this PR are causing the above error lines since there is no mention of OpenSSL in there. 

I found on my successful local Linux build that installed/x64-linux/build-1 exists and there is an apr_rules.mk file in there installed from the apr package per its log.
Could it be that the CI's cached apr package is missing that folder?

**Comment by NancyLi1013** [comment]
Hi @Septarius 
Could you please merge the latest commit on master to this PR?

**Comment by Septarius** [comment]
@NancyLi1013 Merged from upstream/master

File: ports/openssl-uwp/portfile.cmake [hunk 1]
```diff
@@ -1,6 +1,5 @@
-if (NOT VCPKG_TARGET_IS_UWP)
-    message(FATAL_ERROR "${PORT} only supports UWP")
-endif()
+vcpkg_fail_port_install(MESSAGE "${PORT} is only for Windows Universal Platform" ON_TARGET "Linux" "OSX")
+
```
Comment (NancyLi1013) [inline ports/openssl-uwp/portfile.cmake:2]:
```suggestion
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@Septarius 
Thanks for your update.
LGTM now. The current regressions are not related with this PR. So you can ignore them. We will rerun this PR once fixed.

**Comment by BillyONeal** [comment]
>Could it be that the CI's cached apr package is missing that folder?

Yes, that means some other port is playing "is that already present? If not copy over it" games

**Comment by Septarius** [comment]
PR #12067 fixed the build errors. Thanks everyone for the assistance getting this ready!

**Comment by vicroms** [comment]
Thanks for the PR!

**Review by NancyLi1013 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**